app/main.py

```diff
@@ -1,6 +1,21 @@
-from typing import Callable
+from functools import wraps
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Any:
+        serialized_args = str(args) + str(kwargs)
+
+        if serialized_args in cached_results:
+            print("Getting from cache")
+            return cached_results[serialized_args]
+        else:
+            print("Calculating new result")
+            result = func(*args, **kwargs)
+            cached_results[serialized_args] = result
+            return result
```

Comment:
it would look better if you changed the sequence of conditions and used a single return.